### ISSUE TITLE
[EI] Untranslatable texts including macro

### ISSUE THREAD

**Issue by tamanegi-jp** [wesnoth/wesnoth#8225]
### Game and System Information

- **Version: 1.17.24+dev**
- **Downloaded from: github (master)**
- **Build info: built with scons (gcc 13.2)**
- **OS: Linux**

### Description of the bug

Following texts in "Eastern Invasion" contain macros.
Translation does not work for them, since macros are expanded beforehand (unlike variables).
I suppose those macros should be replaced with variables or static strings to enable translation.

scenarios/04b_Ill_Humors.cfg:
_ "To travel east, collect {COUNT} alswdan herbs and pay Hahid al-Ali $gold_cost gold. <i>(challenging)</i>"

scenarios/08_Xenophobia.cfg
scenarios/11_Captured.cfg:
_ "{AMOUNT} gold"

scenarios/08_Xenophobia.cfg:
_ "These {RACE} have a pile of {AMOUNT} gold pieces!"

Note: the race names here (dwarves, serpents, and orcs) are not marked as translatable.
```
    {GOLD_PICKUP 11 4  items/gold-coins-small.png 10 dwarves}
    {GOLD_PICKUP 37 26 items/gold-coins-small.png 20 serpents}
    {GOLD_PICKUP 38 10 items/gold-coins-small.png 20 orcs}
```

scenarios/09_Castle_in_the_Ice.cfg:
_ "Living units take {AMOUNT} cold damage every turn. Damage increases every 10 turns."

scenarios/11_Captured.cfg:
_ "You’ ve pillaged {AMOUNT} orcish gold pieces!"

utils/traits.cfg:
_ "The ‘deserter’ trait has caused ${UNIT}.name to flee back to your recall list."

**Comment by stevecotton** [comment]
The "to travel east" ones were already fixed in #8095, and AFAICS were already fixed in the .po files, but the others weren't caught. Thanks for the bug report.

**Comment by CelticMinstrel** [comment]
The `{RACE}` part needs to be fixed by having a string for each race; probably define a macro for each of those strings and substitute `{RACE}` into the macro name, like `{{RACE}_HAS_GOLD}` or something.

The one in traits.cfg probably needs to be split into a male and female version of the string. I don't know exactly what `{UNIT}` refers to though, so it's possibly it's always a male unit.

**Comment by stevecotton** [comment]
> The one in traits.cfg probably needs to be split into a male and female version of the string. I don't know exactly what `{UNIT}` refers to though, so it's possibly it's always a male unit.

Having just debugged through it, that macro is used in `[event]name=attacker_hits` and `[event]name=defender_hits`, with `{UNIT}` expanding to `$second_unit` or `$unit` respectively. So it needs to both be replaced with something translatable, and it needs male and female forms.

**Comment by stevecotton** [comment]
Assigning to myself because I want to get this fixed in time for the new .pot files to be included in 1.17.25.
